### PR TITLE
Capture variables in tests and add logging.

### PR DESCRIPTION
**Pull request by phbnf**
Capture variable in tests to execute them all. It also adds logging to see how tests have performed.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
My suggestion would be to remove `fixchain/ratelimiter` altogether instead.

File: fixchain/ratelimiter/limiter_test.go [hunk 1]
```diff
@@ -42,13 +43,15 @@ func TestRateLimiterSingleThreaded(t *testing.T) {
 			if qps > float64(limit)*1.01 {
 				t.Errorf("#%d: Too many operations per second. Expected ~%d, got %f", i, limit, qps)
 			}
+			log.Infof("#%d: Expected ~%d, got %f", i, limit, qps)
```
Comment (pavelkalinnikov) [inline fixchain/ratelimiter/limiter_test.go:46]: I think this is redundant. The expected value is part of the test name, and the `qps` only needs printing if it's not matching the expected value, which already happens 2 lines above.

Same below.

File: fixchain/ratelimiter/limiter_test.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ var testlimits = []int{1, 10, 50, 100, 1000}
 func TestRateLimiterSingleThreaded(t *testing.T) {
 	for i, limit := range testlimits {
 		t.Run(fmt.Sprintf("%d ops/s", limit), func(t *testing.T) {
+			i, limit := i, limit
```
Comment (pavelkalinnikov) [inline fixchain/ratelimiter/limiter_test.go:29]: Nice catch.

**Review by pavelkalinnikov — APPROVED** [review]
Approving to not block you, but see the removal suggestion.

**Comment by phbnf** [comment]
+1 for removing them altogether. I have observed unexpected test failures where more than 2k operations per seconds happened, when there should never be more than 1k. Given the unpredictable behaviour of the current test code, I cannot tell for sure if it's because we're using `rate.Limiter` wrong, or if it was just a problem with the test. I would like to see the tests passing first in to gain confidence about our code. Then we can use `rate.Limiter` directly.

**Pull request merged**